rstudio/rsconnect-jupyter
# Option to create manifest and requirements.txt

### Description

This PR adds a button to the publishing dialog to create a manifest and requirements.txt. This enables publishing notebooks via the new git support in Connect.

Additionally, an existing requirements.txt will be honored if present (it won't be overwritten, and will be used in preference to inspecting the environment with `pip freeze`). Currently, the requirements file must include `nbconvert` and `ipykernel` or packages which depend on them in order for rendering to work.

Connected to #15350

### Testing Notes / Validation Steps
* Use the Create Manifest button to create a manifest and requirements file. Try clicking the links to edit them (especially the requirements file). You can also see them in the Jupyter file manager. Deploy, and see that the requirements you specified are included in the requirements build.
* Remove the manifest and create again. It should create the manifest but leave your requirements.txt file alone.
* Push the files (notebook, requirements, manifest) to a git repo and deploy to Connect via git. The deployment should succeed and again honor your requirements.txt file.
